tmt/result.py

```diff
@@ -363,6 +374,24 @@ def interpret_result(self, interpret: ResultInterpret) -> 'Result':
 
         return self
 
+    def to_subresult(self, invocation: 'tmt.steps.execute.TestInvocation') -> 'SubResult':
+        """ Convert result to tmt subresult """
+
+        # TODO: Try to find a better way for conversion (e.g. using
+        # from_spec/to_spec/from_serialized/to_serialized).
+        return SubResult(
+            # Add subresult name under the test namespace
+            name=f"{invocation.test.name}/{self.name.lstrip('/')}",
```

Comment:
Do we want to prefix the subtest name with the main test name? On the level of subtets there should be no collision and showing shorter names in the report seems as a better option from user perspective, I'd say. Currently it looks like this:

```
report
    how: display
        fail /test/beakerlib/bad
            fail /test/beakerlib/bad/Test
        pass /test/beakerlib/good
            pass /test/beakerlib/good/Test
        warn /test/beakerlib/need
            warn /test/beakerlib/need/Cleanup
        errr /test/beakerlib/weird (beakerlib: TestResults FileError)
    summary: 1 test passed, 1 test failed, 1 warn and 1 error
```

It could look like this:

```
report
    how: display
        fail /test/beakerlib/bad
            fail Test
        pass /test/beakerlib/good
            pass Test
        warn /test/beakerlib/need
            warn Cleanup
        errr /test/beakerlib/weird (beakerlib: TestResults FileError)
    summary: 1 test passed, 1 test failed, 1 warn and 1 error
```

Reply:
@juk, any preference or expectations here?

Reply:
+1 for *not* prefixing - I'd keep the phase/subtest name as provided by the test and/or framework.

Reply:
If prefixing is not needed, that's great - it will simplify to code.

Just to be sure - Do we also want to remove the `/` prefix from the **subresult** name? I suppose yes.

```
fail /test/beakerlib/bad
            fail Test
```

vs.

```
fail /test/beakerlib/bad
            fail /Test
```

Reply:
Ok, I've removed the code for prefixing. But the subresult names (both beakerlib and `tmt-report-result`) are **still** prefixed with `/`. This is done by `tmt-report-result` script for "shell" tests and for beakerlib this seems to be an automatic behavior. So, I suppose this answers my question :) - let's keep it on the framework.

```
report
    how: display
        fail /test/beakerlib
            pass /phase-setup
            pass /Internal-test-of-environment-variable-values
            pass /phase-test-pass
            fail /phase-test-fail
            pass /extra-tmt-report-result/good
            fail /extra-tmt-report-result/bad
            warn /extra-tmt-report-result/weird
            skip /extra-tmt-report-result/skip
            pass /phase-test-multiple-tmt-report-result
            pass /phase-cleanup
        fail /test/shell/fail
            pass /fail-subtest/good
            fail /fail-subtest/fail
            warn /fail-subtest/weird
            skip /fail-subtest/skip
        pass /test/shell/pass
            pass /pass-subtest/good0
            pass /pass-subtest/good1
            pass /pass-subtest/good2
        pass /test/shell/skip
            pass /skip-subtest/extra-pass
            skip /skip-subtest/extra-skip
    summary: 2 tests passed and 2 tests failed
```

Reply:
Yeah, looks good. Both main result and subresult names look consistent. And we find this does not please user eye we can always remove later.